From 81347720c2c2ab632c74312614e9549c34fdc2b4 Mon Sep 17 00:00:00 2001
From: Charlie Sharpsteen <source@sharpsteen.net>
Date: Tue, 2 Jul 2019 21:58:55 -0700
Subject: [PATCH 03/12] Remove debian armhf pl-build-tools dependencies

This commit patches various components that go into building puppet-agent
to remove dependencies on packages from `pl-build-tools-vanagon`. The
debian-9-armhf and debian-10-armhf platforms use either system tools and
cross-compilers, or run cross-compiled binaries under QEMU.
---
 configs/components/leatherman.rb     |  9 +++++----
 configs/components/puppet-runtime.rb |  5 ++++-
 configs/components/puppet.rb         |  6 ++++--
 configs/components/runtime.rb        | 13 ++++++++-----
 4 files changed, 21 insertions(+), 12 deletions(-)

diff --git a/configs/components/leatherman.rb b/configs/components/leatherman.rb
index c30ac931..58d375f3 100644
--- a/configs/components/leatherman.rb
+++ b/configs/components/leatherman.rb
@@ -3,7 +3,11 @@ component "leatherman" do |pkg, settings, platform|
 
   make = platform[:make]
 
-  if platform.is_macos?
+  if platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/ ||
+        (platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/)
+    # These platforms use their default OS toolchain and have package
+    # dependencies configured in the platform provisioning step.
+  elsif platform.is_macos?
     pkg.build_requires "cmake"
     pkg.build_requires "gettext"
   elsif platform.name =~ /solaris-10/
@@ -18,9 +22,6 @@ component "leatherman" do |pkg, settings, platform|
     pkg.build_requires "cmake"
     pkg.build_requires "pl-toolchain-#{platform.architecture}"
     pkg.build_requires "pl-gettext-#{platform.architecture}"
-  elsif platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/
-    # These platforms use their default OS toolchain and have package
-    # dependencies configured in the platform provisioning step.
   else
     pkg.build_requires "pl-cmake"
     pkg.build_requires "pl-gettext"
diff --git a/configs/components/puppet-runtime.rb b/configs/components/puppet-runtime.rb
index 6a6087e0..a491c593 100644
--- a/configs/components/puppet-runtime.rb
+++ b/configs/components/puppet-runtime.rb
@@ -44,7 +44,8 @@ component 'puppet-runtime' do |pkg, settings, platform|
 
   pkg.install_only true
 
-  if platform.is_cross_compiled_linux? || platform.is_solaris? || platform.is_aix?
+  if !(platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/) &&
+      (platform.is_cross_compiled_linux? || platform.is_solaris? || platform.is_aix?)
     pkg.build_requires 'runtime'
   end
 
@@ -54,6 +55,8 @@ component 'puppet-runtime' do |pkg, settings, platform|
     if platform.os_version == "11"
       pkg.build_requires 'pl-ruby'
     end
+  elsif platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/
+    # We use puppet-runtime's Ruby via QEMU
   elsif platform.is_cross_compiled_linux?
     pkg.build_requires 'pl-ruby'
   end
diff --git a/configs/components/puppet.rb b/configs/components/puppet.rb
index ea35a1ae..829df7e5 100644
--- a/configs/components/puppet.rb
+++ b/configs/components/puppet.rb
@@ -12,7 +12,8 @@ component "puppet" do |pkg, settings, platform|
     pkg.build_requires "pl-gettext-#{platform.architecture}"
   elsif platform.is_aix?
     pkg.build_requires "http://pl-build-tools.delivery.puppetlabs.net/aix/#{platform.os_version}/ppc/pl-gettext-0.19.8-2.aix#{platform.os_version}.ppc.rpm"
-  elsif platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/
+  elsif platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/ ||
+        (platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/)
     # These platforms use their default OS toolchain and have package
     # dependencies configured in the platform provisioning step.
   elsif !platform.is_solaris?
@@ -110,7 +111,8 @@ component "puppet" do |pkg, settings, platform|
       msgfmt = "/cygdrive/c/tools/pl-build-tools/bin/msgfmt.exe"
     elsif platform.is_macos?
       msgfmt = "/usr/local/opt/gettext/bin/msgfmt"
-    elsif platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/
+    elsif platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/ ||
+          (platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/)
       msgfmt = "msgfmt"
     else
       msgfmt = "/opt/pl-build-tools/bin/msgfmt"
diff --git a/configs/components/runtime.rb b/configs/components/runtime.rb
index db46f132..dba1d719 100644
--- a/configs/components/runtime.rb
+++ b/configs/components/runtime.rb
@@ -2,7 +2,12 @@
 component "runtime" do |pkg, settings, platform|
   pkg.add_source "file://resources/files/runtime/runtime.sh"
 
-  if platform.name =~ /solaris-10/
+  if platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/ ||
+        platform.is_macos? ||
+        (platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/)
+    # These platforms use their default OS toolchain and have package
+    # dependencies configured in the platform provisioning step.
+  elsif platform.name =~ /solaris-10/
     if platform.architecture == 'sparc'
       pkg.build_requires "http://pl-build-tools.delivery.puppetlabs.net/solaris/10/pl-gcc-4.8.2-9.#{platform.architecture}.pkg.gz"
       pkg.build_requires "http://pl-build-tools.delivery.puppetlabs.net/solaris/10/pl-binutils-2.27-2.#{platform.architecture}.pkg.gz"
@@ -24,9 +29,6 @@ component "runtime" do |pkg, settings, platform|
     pkg.build_requires "pl-pdcurses-#{platform.architecture}"
     # We only need zlib because curl is dynamically linking against zlib
     pkg.build_requires "pl-zlib-#{platform.architecture}"
-  elsif platform.name =~ /sles-15|fedora-(29|30)|el-8|debian-10/ || platform.is_macos?
-    # These platforms use their default OS toolchain and have package
-    # dependencies configured in the platform provisioning step.
   else
     pkg.build_requires "pl-gcc"
   end
@@ -46,7 +48,8 @@ component "runtime" do |pkg, settings, platform|
   if platform.is_aix?
     pkg.install_file File.join(libdir, "libstdc++.a"), "/opt/puppetlabs/puppet/lib/libstdc++.a"
     pkg.install_file File.join(libdir, "libgcc_s.a"), "/opt/puppetlabs/puppet/lib/libgcc_s.a"
-  elsif platform.is_macos?
+  elsif platform.is_macos? ||
+        (platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/)
     # Nothing to see here
   elsif platform.is_windows?
     lib_type = platform.architecture == "x64" ? "seh" : "sjlj"
-- 
2.20.1 (Apple Git-117)

